crates/rooch-framework/sources/nft/collection.move

```diff
@@ -0,0 +1,291 @@
+        collection_mut_ref.supply.current = collection_mut_ref.supply.current - 1;
+        if(option::is_some(&collection_mut_ref.supply.maximum)){
+            option::some(collection_mut_ref.supply.current)
+        }else{
+            option::none<u64>()
+        }
+    }
+
+    // assert
+    public fun assert_collection_exist_of_ref(collectionRef: &ObjectRef<Collection>){
+        assert!( object_ref::exist_object(collectionRef), ECollectionNotExist);
+    }
+
+    public fun assert_collection_exist_of_id(collectionID: ObjectID, ctx: & Context){
+        assert!( context::exist_object(ctx, collectionID), ECollectionNotExist);
+        context::borrow_object<Collection>(ctx,collectionID);
+    }
+
+    public fun assert_mutator_exist_of_ref(mutatorRef: &ObjectRef<MutatorRef>){
+        assert!( object_ref::exist_object(mutatorRef), EMutatorNotExist);
+    }
+
+    public fun assert_mutator_exist_of_id(mutatorID: ObjectID, ctx: & Context){
+        assert!( context::exist_object(ctx, mutatorID), EMutatorNotExist);
+        context::borrow_object<MutatorRef>(ctx, mutatorID);
+    }
+
+    public fun add_display(mutator: &ObjectRef<MutatorRef>, display: Display, ctx: &mut Context){
+        add_extend_internal(mutator, display, ctx);
+    }
+
+    public fun borrow_display(mutator: &ObjectRef<MutatorRef>, ctx: &mut Context):&Display{
+        borrow_extend_internal(mutator, ctx)
+    }
+
+    public fun borrow_mut_display(mutator: &ObjectRef<MutatorRef>, ctx: &mut Context):&mut Display{
+        borrow_mut_extend_internal(mutator, ctx)
+    }
+
+    public fun remove_display(mutator: &ObjectRef<MutatorRef>, ctx: &mut Context):Display{
+        remove_extend_internal(mutator, ctx)
+    }
+
+    public fun contains_display(mutator: &ObjectRef<MutatorRef>, ctx: &mut Context): bool{
+        contains_extend_internal<Display>(mutator, ctx)
+    }
+
+    #[private_generics(V)]
+    public fun add_extend<V: key>(mutator: &ObjectRef<MutatorRef>, val: V, ctx: &mut Context){
+        add_extend_internal(mutator, val, ctx);
+    }
+
+    public fun borrow_extend<V: key>(mutator: &ObjectRef<MutatorRef>, ctx: &mut Context):&V{
+        borrow_extend_internal(mutator, ctx)
+    }
+
+    #[private_generics(V)]
+    public fun borrow_mut_extend<V: key>(mutator: &ObjectRef<MutatorRef>, ctx: &mut Context):&mut V{
+        borrow_mut_extend_internal(mutator, ctx)
+    }
+
+    #[private_generics(V)]
+    public fun remove_extend<V: key>(mutator: &ObjectRef<MutatorRef>, ctx: &mut Context):V{
+        remove_extend_internal(mutator, ctx)
+    }
+
+    public fun contains_extend<V: key>(mutator: &ObjectRef<MutatorRef>, ctx: &mut Context): bool{
+        contains_extend_internal<V>(mutator, ctx)
+    }
+
+
+    fun add_extend_internal<V: key>(mutator: &ObjectRef<MutatorRef>,val: V,ctx: &mut Context){
+        assert_mutator_exist_of_ref(mutator);
+        let mutator_object_ref = object_ref::borrow(mutator);
```

Comment:
@jolestar  Currently an error is generated here and it looks like it was not successfully created.
│ 154 │     native fun borrow_box<K: copy + drop, V, B>(table_handle: ObjectID, key: K): &Box<V>;
│     │                ^^^^^^^^^^
│     │                │
│     │                Test was not expected to error, but it aborted with code 393218 originating in the module 0000000000000000000000000000000000000000000000000000000000000002::raw_table rooted here
│     │                In this function in 0x2::raw_table